src/services/armService.ts

```diff
@@ -148,17 +142,58 @@ export class ArmService extends BaseService {
     }
   }
 
+  private compareDeployments(current: ArmDeployment, previous: ArmDeployment, valuedParameters: any): boolean {
```

Comment:
Consider renaming function. `compareDeployments` is not quickly identifiable as returning `boolean` and also I wouldn't know if getting a value back as `true` meant i needed to redeploy or that they are equal.

How about `areDeploymentsEqual` ?